crates/loader/src/local/config.rs

```diff
@@ -23,11 +24,57 @@ pub(crate) type RawComponentManifestPartial = RawComponentManifestImpl<toml::Val
 
 /// Container for any version of the manifest.
 #[derive(Clone, Debug, Deserialize, Serialize)]
-#[serde(tag = "spin_version")]
+#[serde(untagged)]
 pub enum RawAppManifestAnyVersionImpl<C> {
-    /// A manifest with API version 1.
-    #[serde(rename = "1")]
-    V1(RawAppManifestImpl<C>),
+    /// Old Spin manifest versioned with `spin_version` key
+    V1Old {
+        #[serde(deserialize_with = "deserialize_fixed_version_1")]
+        /// Version key name
+        spin_version: spin_app::locked::FixedVersion<1>,
+        /// Manifest
+        #[serde(flatten)]
+        manifest: RawAppManifestImpl<C>,
+    },
+    /// New Spin manifest versioned with `spin_manifest_version` key
+    V1New {
+        #[serde(deserialize_with = "deserialize_fixed_version_1")]
+        /// Version key name
+        spin_manifest_version: spin_app::locked::FixedVersion<1>,
+        /// Manifest
+        #[serde(flatten)]
+        manifest: RawAppManifestImpl<C>,
+    },
+}
+
+fn deserialize_fixed_version_1<'de, D>(deserializer: D) -> Result<FixedVersion<1>, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+{
+    let s: String = serde::de::Deserialize::deserialize(deserializer)?;
+    FixedVersion::try_from(s).map_err(serde::de::Error::custom)
+}
+
+impl<C> RawAppManifestAnyVersionImpl<C> {
+    /// Get V1 manifest
+    pub fn try_into_v1(self) -> Option<RawAppManifestImpl<C>> {
+        Some(match self {
+            RawAppManifestAnyVersionImpl::V1New { manifest, .. } => manifest,
+            RawAppManifestAnyVersionImpl::V1Old { manifest, .. } => manifest,
+        })
+    }
+
+    /// Get V1 manifest
+    pub fn into_v1(self) -> RawAppManifestImpl<C> {
```

Comment:
I am dubious about the implementation - it looks like the sort of thing that could accidentally get used on odd places and blow up at runtime if/when there is a v2.  If we have this function (and I appreciate we don't want to repeat both patterns in multiple places), I'd like it to do the pattern match so that when v2 comes along the compiler shouts at us and we have to grapple with a safe way of addressing it and hopefully are forced to look mindfully at the sites where this is used.

Reply:
Ivan, I generally agree. But I'd argue that when we introduce V2 we would have revisited and consolidated these serde datastructures in a uniform way. 

FWIW this isn't an uncommon pattern used in rust code. You lose compile time type safety but the idea is that if you need to check if v1 or v2 you'd pattern match only called into_v1 in guaranteed infallible situations

Reply:
Ideally once we're moving to a V2 any differences between V1 and V2 would be hidden in some wrapper. This pattern is fine when needed but YAGNI here I think.

Reply:
The problem is situations which are infallible now but become fallible later, though, or am I missing your point?  How do you know where it is safe to call this?

Reply:
I agree. But for 1.0 time frame this seems sufficient.

Reply:
Agree but this is the stage where we are dealing with the raw form before we wrap it.

Sounds like others are comfortable though so I'll recede into my hedge.

Reply:
@ivan I agree completely with your concern. This is just a bandaid for technical debt we'll need to address for a more incisive refactoring to anticipate v2.

Reply:
We added an `Option` wrapper around `try_into_v1` for flexibility when v2 comes, but it sounds like we will need a refactor when v2 comes anyways. I removed the Option so there isn't any unnecessary unwrapping. It seems pointless to anticipate what we will need for v2 since we will likely be making another wrapper.